src/wasm_standalone.cmake

```diff
@@ -9,12 +9,28 @@ set(CMAKE_SYSTEM_PROCESSOR wasm32)
 set(triple wasm32-unknown-unknown-wasm)
 
 set(WASM_SDKROOT ${CMAKE_CURRENT_LIST_DIR})
-set(CMAKE_C_COMPILER ${WASM_SDKROOT}/bin/clang)
-set(CMAKE_CXX_COMPILER ${WASM_SDKROOT}/bin/clang++)
-set(CMAKE_AR ${WASM_SDKROOT}/bin/llvm-ar)
-set(CMAKE_RANLIB ${WASM_SDKROOT}/bin/llvm-ranlib)
+
+if (CMAKE_HOST_WIN32)
+        set(EXE_SUFFIX ".exe")
+else()
+        set(EXE_SUFFIX "")
+endif()
+if ("${CMAKE_C_COMPILER}" STREQUAL "")
+  set(CMAKE_C_COMPILER ${WASM_SDKROOT}/bin/clang${EXE_SUFFIX})
+endif()
+if ("${CMAKE_CXX_COMPILER}" STREQUAL "")
+  set(CMAKE_CXX_COMPILER ${WASM_SDKROOT}/bin/clang++${EXE_SUFFIX})
+endif()
+if ("${CMAKE_AR}" STREQUAL "")
+  set(CMAKE_AR ${WASM_SDKROOT}/bin/llvm-ar${EXE_SUFFIX} CACHE FILEPATH "llvm ar")
+endif()
+if ("${CMAKE_RANLIB}" STREQUAL "")
+ set(CMAKE_RANLIB ${WASM_SDKROOT}/bin/llvm-ranlib${EXE_SUFFIX} CACHE FILEPATH "llvm ranlib")
+endif()
 set(CMAKE_C_COMPILER_TARGET ${triple})
 set(CMAKE_CXX_COMPILER_TARGET ${triple})
+set(CMAKE_C_FLAGS --target=${triple})
+set(CMAKE_CXX_FLAGS --target=${triple})
```

Comment:
Why wasn't this needed before?

Reply:
It affects CMake's detection of the compiler's default target. If you don't put that, CMake thinks the compiler is targeting Windows.

Reply:
(anyway, I think it was needed before; we just didn't have it. We don't want to require the user to pass --target themselves). What I think we should actually do is make a symlink to name the compiler `wasm32-clang` as other gnu-style cross-compilers do, and then it will automatically get the target right. We'll just have to coordinate that with an emscripten change.